debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlConnectorConfig.java

```
@@ -1014,6 +1029,29 @@ private static int validateEventDeserializationFailureHandlingModeNotSet(Configu
         return 0;
     }
 
+    private JdbcCredentialsProvider getCredentialsProvider(Configuration config){
+        String providerClass = config.getString(MySqlConnectorConfig.CREDENTIALS_PROVIDER_CLASS_NAME);
+        LOGGER.info("Credentials provider class: {}", providerClass);
+        LOGGER.info("DEBUGIAMASSUMEROLE -Creating credentials provider: {}", providerClass);
+        try {
+            LOGGER.info("DEBUGIAMASSUMEROLE - Current config : {}", config.asMap());
+            JdbcCredentialsProvider provider = (JdbcCredentialsProvider) Class.forName(providerClass)
+                    .getDeclaredConstructor().newInstance();
+
+            LOGGER.info("Successfully created a new instance of credentials provider of type: {}", providerClass);
+            LOGGER.info("DEBUGIAMASSUMEROLE -Configuring credentials provider with config: {}", config.asMap());
+            provider.configure(config.asMap());
+            LOGGER.info("DEBUGIAMASSUMEROLE -Configured credentials provider: {}", provider);
+
+            return provider;
+        }
+        catch (Exception e) {
+            LOGGER.warn("Error initializing credentials provider {}: {}", providerClass, e.getMessage());
+            LOGGER.debug("Detailed provider initialization error", e);
+            return null;
```

We should re-throw the exception here. Since we are returning null, in future it might lead to irrelevant hard to debug NPEs when we use the value returned from this function.

---

Done. Can you check ?